### DISCUSSION TITLE
Is there `cuda::aligned_size_t` on pre-Volta devices ?

### DISCUSSION THREAD

**Discussion by fkallen** [NVIDIA/libcudacxx#374] [General]
```
cooperative_groups::memcpy_async(
  const TyGroup &group,
  TyElem *__restrict__ dst,
  const DstLayout &dstLayout,
  const TyElem *__restrict__ src,
  const SrcLayout &srcLayout
);
```
 requires `Compute Capability 3.5 minimum, Compute Capability 8.0 for asynchronicity, C++11`

cuda::aligned_size_t is only defined in <cuda/barrier> and <cuda/pipeline> which are only available for CC >= 7.0.
```
#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ < 700
#  error "CUDA synchronization primitives are only supported for sm_70 and up."
#endif
```

To use aligned_size_t on older devices, one needs to manually define aligned_size_t .

Problem example: https://cuda.godbolt.org/z/zv5v15z5W

**Comment by jrhemstad** [comment]
It is not currently available because of the header it is in. I created an issue for this a while back: https://github.com/NVIDIA/libcudacxx/issues/290
